templates/html/.filters/all.js

```diff
@@ -117,8 +117,18 @@ module.exports = ({ Nunjucks, Markdown, OpenAPISampler }) => {
   });
 
   Nunjucks.addFilter('getPayloadExamples', (msg) => {
-    if (Array.isArray(msg.examples()) && msg.find(e => e.payload)) {
-      return msg.filter(e => e.payload);
+    if (Array.isArray(msg.examples()) && msg.examples().find(e => e.payload)) {
+      const ex = msg.examples().filter(e => e.payload);
+      if (ex) {
+        //Go through each found payload from filtering the examples array.
+        let returnArray = [];
+        for (var i in ex) {
+          returnArray = returnArray.concat(ex[i].payload);
+        }
+        return returnArray;
+      } else {
+        return undefined;
+      }
```

Comment:
I think you can probably replace all this code with this:

```js
return msg.examples().map(e => e.payload).filter(Boolean);
```

The `.filter(Boolean)` trick will remove all the `undefined` items in the array.

Reply:
Not a bad little 1 liner, problem still remains that we would need to use something like [flatMap ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat) which are only supported in node >= 11 to flatten the array. Instead I used their alternative example provided and ended up with:
`return msg.examples().map(e => e.payload).filter(Boolean).reduce((acc, val) => acc.concat(val), []);`

The checks are still needed in order to ensure payload examples are used (if any) when message examples are not defined.